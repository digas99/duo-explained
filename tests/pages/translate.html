<!DOCTYPE html>
<html data-duo-theme="dark" dir="ltr" lang="en"><head><title data-react-helmet="true">Duolingo - The world's best way to learn French</title>     </head><body style="--onetrust-banner-height: 0px;"><div class="fs-unmask" id="root" onclick=""><div class="kPqwA _2kkzG"><div class="_3yE3H"><div class="_2G9ED _29gfw"><div class="wqSzE"><div class="_3v4ux"><div class="I-Avc _1zcW8"><button class="_3vGNs _2YF0P _3X5b9" data-test="quit-button"><img class="_3X5b9 _1o-YO" src="./translate_files/df223d5b9feb8017b323ed21103eb5ac.svg"/></button><div aria-valuemax="1" aria-valuemin="0" aria-valuenow="0.058823529411764705" class="_358Nq" role="progressbar" style="--web-ui_progress-bar-color: rgb(var(--color-owl)); --web-ui_progress-bar-shine-height: 3px; --__internal__progress-bar-height: 16px; --__internal__progress-bar-inner-value: 0%; --__internal__progress-bar-value: 5.88%;"><div class="_2-LJi"><div class="_2fuvy" style="opacity: 1;"><div class="_3Dzp2 _2fuvy"></div><div class="_1EzVJ"></div></div><div class="_3FEy7" style="opacity: 1;"><div class="_2fYtx"><span style="--progress-bar-height: 16px;"><span class="RxQcJ fs-exclude SVDhH"></span></span></div></div></div></div><div class="_3ww7z"><img src="./translate_files/0c1a523b4b5882a97e4df162f4b5c58b.svg" style="height: 32px; width: 32px;"/><img src="./translate_files/8d4c5fcd9cedabfd155d3eda8af269bc.svg"/></div></div></div></div><div class="RMEuZ _1GVfY"><div class="_3GuWo _1cTBC"><div class="_1XNQX"><div class="_1fxa4 _1Mopf" data-test="challenge challenge-translate"><div class="_2n5fx _1JTA4 _1es1Y K3vbJ"><div class="U41Ye"><h1 class="_3EOK0" data-test="challenge-header"><span>Write this in English</span></h1></div><div class="iPZZY UjFh4 _3rat3"><div class="_2w0y3 _1dtTU _35mGI"><div class="_31yjb"><div class="_2qg6J"><div class="" style="width: 100%; height: 100%; transform: scale(1.25) translate(-6.94444%, -8.52273%);"><canvas height="175" style="vertical-align: top; width: 118px; height: 175px;" width="118"></canvas></div></div><div class="rPXvv _44BHt"><div class="_1RKXT"><div class="_2L10X" dir="ltr" lang="fr"><span class="_12Ih_"><span class="_3KG7r _1fdKO" dir="ltr"><button class="_1GJVt _3DAip bafGS _2LoNU VzbUl _1saKQ _1AgKJ"><span class="RxQcJ fs-exclude _1OCYa" style="--animated-speaker-icon-color: rgb(var(--color-macaw));"></span></button></span></span><span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">T</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">u</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">d</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">à</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">N</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">w</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">Y</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">k</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">.</span><div aria-label="Tu" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 0px; top: 0px; width: 20.0938px;" tabindex="0"></div><div aria-label="dois" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 24.9375px; top: 0px; width: 35.875px;" tabindex="0"></div><div aria-label="aller" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 65.6562px; top: 0px; width: 40.75px;" tabindex="0"></div><div aria-label="à" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 111.25px; top: 0px; width: 10.4062px;" tabindex="0"></div><div aria-label="New" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 126.5px; top: 0px; width: 38.25px;" tabindex="0"></div><div aria-label="York" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 28px; left: 169.344px; top: 0px; width: 38.2344px;" tabindex="0"></div></span></span></div></div></div></div></div><div class="u5Wzl _32LPr _3hg-V"><div class="Sa7Uw"><div class="MvChQ v1KUv fUvcy"><div class="eWdJ5"><div class="_32DLo"><div class="_1VxfZ" dir="ltr"><div class="_1arte"><div class="_3gJ3o D0fy9"></div><div class="_3gJ3o D0fy9"></div><div class="_3gJ3o D0fy9"></div><div class="_3gJ3o D0fy9"></div><div class="_3gJ3o D0fy9"></div></div><div class="_2-F7v"><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: 0;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="You-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">You</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -1;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="have-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">have</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -2;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="to-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">to</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -3;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="go-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">go</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -4;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="to-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">to</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -5;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="New-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">New</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px; z-index: -6;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="York-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">York</span></span></button></span></div></div></div></div></div><div class="_1v1Bd"><div class="eSgkc" data-test="word-bank"><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="after-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">after</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="for-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">for</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="to-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">to</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="go-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">go</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="streets-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">streets</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="You-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">You</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="have-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">have</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="York-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">York</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q WxjqG _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX" data-test="tree-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">tree</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="New-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">New</span></span></button></span></div><div class="_1uV0Q _DVHp" style="--tap-tokens-token-margin-before: 4px; --tap-tokens-token-margin-after: 4px;"><span class="_3VyQa"><button aria-disabled="true" class="_1NM1Q _1Qh5D _36g4N _2YF0P notranslate _3ZtW_ _2O7Ua _3U5_i _1HvEX _31wR2" data-test="to-challenge-tap-token" dir="ltr" lang="en" translate="no"><span class="_12ozk"></span><span class="_231NG"><span data-test="challenge-tap-token-text">to</span></span></button></span></div></div></div></div></div></div></div></div></div></div></div></div><div class="_3GuWo _1cTBC _1QQhE"><div class="_1XNQX"><div class="_3FB5S _1VTif _3yMvO _2HXQ9 d-cgpt-footer" id="session/PlayerFooter"><div class="U8jH3 jHbiF"><div class="AXBVw _2Ycbe _8wIx-"><div class="_1k6eg" data-test="blame blame-correct"><div class="_31ixh HJdJI fwpkb"><img alt="" class="_3oCTd" src="./translate_files/b377ec812acb8c96d87d52e8009478ad.svg"/></div><div class="d6Xhl"><div class="_3LKiy"><div class="_1D3fo"><h2 class="_2U7Gm WXwlk">Good job!</h2></div></div><div class="_2yBiG"><div class="_10y_e"><button class="_1pDd6 _2JTJI bafGS _2LoNU VzbUl _1saKQ _1AgKJ"><span class="_24Rh2"><span class="_2e3N3 _3jt-c"></span><span class="_1qfbO _2Rt1l">Report</span></span></button></div></div></div></div></div><div class="MYehf d-cgpt-explain-button-wrapper"><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR d-cgpt-explain-button d-cgpt-explain-button-enabled" data-hasclick="true" id="d-cgpt-explain-button"><span class="_1o-YO">Explain</span></button><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR" data-test="player-next"><span class="_1o-YO">Continue</span></button></div></div></div>
<div class="d-cgpt-speech-bubble" style="bottom: 130px; display: none;">
<div>
<textarea placeholder="If you want, add to the query to ChatGPT..."></textarea>
<div class="d-cgpt-speech-bubble-tip"></div>
</div>
</div>
</div></div></div></div></div><div class="Vm8CO _1Fnem" data-test="drawer-backdrop" style="pointer-events: none;"></div></div><div class="fs-unmask" id="overlays" onclick=""><div class="Vm8CO _2zxQ8" style="pointer-events: none;"></div><a class="_1Qh5D _36g4N _2YF0P _3dDzz _1wiIJ" href="https://preview.duolingo.com/help/support-request" rel="noopener" target="_blank" title="Report feedback"><img class="_3QUr-" src="./translate_files/b23d90ca7ea01eb330158d4939dc4899.svg"/></a><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div><div data-focus-lock-disabled="disabled"></div><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div></div><div><div class="grecaptcha-badge" data-style="bottomright" style="width: 256px; height: 60px; display: block; transition: right 0.3s; position: fixed; bottom: 14px; right: -186px; box-shadow: gray 0px 0px 5px; border-radius: 2px; overflow: hidden;"><div class="grecaptcha-logo"></div><div class="grecaptcha-error"></div><textarea class="g-recaptcha-response" id="g-recaptcha-response-100000" name="g-recaptcha-response" style="width: 250px; height: 40px; border: 1px solid rgb(193, 193, 193); margin: 10px 25px; padding: 0px; resize: none; display: none;"></textarea></div></div><div id="onetrust-consent-sdk"><div class="onetrust-pc-dark-filter ot-hide ot-fade-in"></div></div></body></html>