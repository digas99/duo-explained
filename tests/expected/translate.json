{
	"content": {
		"sentence": "Tu dois aller Ã  New York.",
		"answer": [
			"You",
			"have",
			"to",
			"go",
			"to",
			"New",
			"York"
		],
		"wordBank": [
			"after",
			"for",
			"to",
			"go",
			"streets",
			"You",
			"have",
			"York",
			"tree",
			"New",
			"to"
		]
	},
	"exercise": "Write this in English",
	"wrapper": {},
	"type": "translate"
}